### PR TITLE
Core 145 - Ethers contract estimateGas and populateTransaction for overriding functions

### PR DESCRIPTION
**Pull request by livingrock7**

### CODE REVIEW

File: src/Biconomy.js [hunk 1]
```diff
@@ -321,10 +321,13 @@ Biconomy.prototype.getForwardRequestAndMessageToSign = function (
         }
         _logMessage("API found");
         let params = methodInfo.params;
+        let typeString = "";
         let paramArray = [];
         for (let i = 0; i < params.length; i++) {
           paramArray.push(_getParamValue(params[i]));
+          typeString = typeString + params[i].type.toString()+",";
         }
+        typeString = typeString.substring(0,typeString.length - 1);
```
Comment (tomarsachin2271) [inline src/Biconomy.js:330]: What will this return if params.length is 0?

File: src/Biconomy.js [hunk 2]
```diff
@@ -630,17 +634,21 @@ async function sendSignedTransaction(engine, payload, end) {
...
 
             let paramArrayForGasCalculation = [];
+            let typeString = "";
             for (let i = 0; i < params.length; i++) {
               paramArrayForGasCalculation.push(
                 _getParamValue(params[i])
               );
+              typeString = typeString + params[i].type.toString()+",";
             }
+            typeString = typeString.substring(0,typeString.length - 1);
```
Comment (tomarsachin2271) [inline src/Biconomy.js:644]: What this will return if params.length = 0 ?

File: src/Biconomy.js (hunk 1 above)
Comment (livingrock7) [reply src/Biconomy.js:330]: this should be enough

if(params.length > 0) {
        typeString = typeString.substring(0,typeString.length - 1);
        }

File: src/Biconomy.js (hunk 2 above)
Comment (livingrock7) [reply src/Biconomy.js:644]: this should be neough

`if(params.length > 0) {
        typeString = typeString.substring(0,typeString.length - 1);
        }`

**Review by tomarsachin2271 — APPROVED** [review]

**Pull request merged**